java-checks/src/main/java/org/sonar/java/checks/UnconditionalJumpStatementCheck.java

```diff
@@ -0,0 +1,237 @@
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.ReturnStatementTree;
+import org.sonar.plugins.java.api.tree.SyntaxToken;
+import org.sonar.plugins.java.api.tree.ThrowStatementTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.WhileStatementTree;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Rule(key = "S1751")
+public class UnconditionalJumpStatementCheck extends IssuableSubscriptionVisitor {
+
+  private static final MethodMatcherCollection NEXT_ELEMENT = MethodMatcherCollection.create(
+    MethodMatcher.create().typeDefinition(TypeCriteria.subtypeOf("java.util.Enumeration")).name("hasMoreElements").withoutParameter(),
+    MethodMatcher.create().typeDefinition(TypeCriteria.subtypeOf("java.util.Iterator")).name("hasNext").withoutParameter());
+
+  private static final Tree.Kind[] LOOP_KINDS = {
+    Tree.Kind.DO_STATEMENT,
+    Tree.Kind.WHILE_STATEMENT,
+    Tree.Kind.FOR_STATEMENT,
+    Tree.Kind.FOR_EACH_STATEMENT
+  };
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(
+      Tree.Kind.BREAK_STATEMENT,
+      Tree.Kind.CONTINUE_STATEMENT,
+      Tree.Kind.RETURN_STATEMENT,
+      Tree.Kind.THROW_STATEMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    Tree parent = tree.parent();
+    while (parent.is(Tree.Kind.BLOCK)) {
+      parent = parent.parent();
+    }
+
+    if (!parent.is(LOOP_KINDS)) {
+      return;
+    }
+
+    if (tree.is(Tree.Kind.CONTINUE_STATEMENT)
+      || (!isWhileNextElementLoop(parent) && !isEmptyConditionLoop(parent) && executeUnconditionnally(parent))) {
+      SyntaxToken jumpKeyword = jumpKeyword(tree);
+      reportIssue(jumpKeyword, String.format("Remove this \"%s\" statement or make it conditional.", jumpKeyword.text()));
+    }
+  }
+
+  /**
+   * For or while loops are sometimes used as short-cut simulating goto's, when having always true conditions:
+   * <code>
+   * for(;;) {
+   *  if (...) {
+   *    // ...
+   *    break;
+   *  }
+   *  while (...) {
+   *    // ...
+   *    if (...) {
+   *      break;
+   *    }
+   *  }
+   *  break; // last unconditional jump to exit the infinite loop
+   * }
+   * </code>
+   */
+  private static boolean isEmptyConditionLoop(Tree loopTree) {
+    switch (loopTree.kind()) {
+      case FOR_STATEMENT:
+        ForStatementTree fst = (ForStatementTree) loopTree;
+        return fst.initializer().isEmpty() && fst.condition() == null && fst.update().isEmpty();
+      case WHILE_STATEMENT:
+        // 'while(false)' does not compile, unreachable code
+        return isTrue(((WhileStatementTree) loopTree).condition());
+      case DO_STATEMENT:
+        // only target true literal, which needs a break statement.
+        // For a 'do {...} while (false);', loop the jump is useless and issue should be raised.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)